shell/platform/fuchsia/flutter/pointer_injector_delegate.cc

```diff
@@ -0,0 +1,318 @@
+    if (!pointer_y->value.IsFloat() && !pointer_y->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'Pointer.Y' is not a float";
+      return false;
+    }
+
+    auto pointer_id = args.FindMember("pointerId");
+    if (!pointer_id->value.IsUint()) {
+      FML_LOG(ERROR) << "Argument 'pointerId' is not a uint32";
+      return false;
+    }
+
+    auto trace_flow_id = args.FindMember("traceFlowId");
+    if (!trace_flow_id->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'traceFlowId' is not a int";
+      return false;
+    }
+
+    auto view_ref = args.FindMember("viewRef");
+    if (!view_ref->value.IsUint64()) {
+      FML_LOG(ERROR) << "Argument 'viewRef' is not a uint64";
+      return false;
+    }
+
+    zx_handle_t handle = view_ref->value.GetUint64();
+    zx_handle_t out_handle;
+    zx_status_t status =
+        zx_handle_duplicate(handle, ZX_RIGHT_SAME_RIGHTS, &out_handle);
+    if (status != ZX_OK) {
+      FML_LOG(ERROR) << "Argument 'viewRef' is not valid";
+      return false;
+    }
+    auto ref = fuv_ViewRef({
+        .reference = zx::eventpair(out_handle),
+    });
+
+    auto width = args.FindMember("logicalWidth");
+    if (!width->value.IsFloat() && !width->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'logicalWidth' is not a float";
+      return false;
+    }
+
+    auto height = args.FindMember("logicalHeight");
+    if (!height->value.IsFloat() && !height->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'logicalHeight' is not a float";
+      return false;
+    }
+
+    auto timestamp = args.FindMember("timestamp");
+    if (!timestamp->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'timestamp' is not a int";
+      return false;
+    }
+
+    PointerInjectorRequest request = {
+        .x = pointer_x->value.GetFloat(),
+        .y = pointer_y->value.GetFloat(),
+        .pointer_id = pointer_id->value.GetUint(),
+        .phase = static_cast<fup_EventPhase>(phase->value.GetInt()),
+        .trace_flow_id = trace_flow_id->value.GetUint64(),
+        .view_ref = std::move(ref),
+        .logical_size = {width->value.GetFloat(), height->value.GetFloat()},
+        .timestamp = timestamp->value.GetInt()};
+
+    // Inject the pointer event if the view has been created. Otherwise buffer
+    // it so that it can be dispatched later.
+    if (valid_views_.count(id) > 0) {
+      valid_views_.at(id).InjectEvent(std::move(request));
+    } else {
+      pending_inject_requests_[id].push(std::move(request));
+    }
+    Complete(std::move(response), "[0]");
+  } else {
+    return false;
+  }
+  // All of our methods complete the platform message response.
+  return true;
+}
+
+void PointerInjectorDelegate::OnCreateView(uint64_t view_id) {
+  FML_CHECK(valid_views_.count(view_id) == 0);
+
+  auto [_, success] =
+      valid_views_.try_emplace(view_id, registry_, host_view_ref_);
+
+  FML_CHECK(success);
+
+  // Dispatch any pending pointer events buffered till now for the view.
```

Comment:
there is no need; Jaeheon advised that we drop any events received before:
1. View is created
2. ViewRef is available
3. (for gfx) ViewStateChanged for that View becomes true for the first time

Reply:
Done